ranges.cpp

```diff
@@ -484,6 +507,214 @@ constexpr auto transform(Func p) {
 }
 /******************************************/
 
+/******************************************/
+/* startAt_view                         */
+/* Iterates over all elements of a view   */
+/* starting in the first element which   */
+/* matches the function, and ending in	  */
+/* the element just before the matching one    */
+/******************************************/
+template<typename View>
+class startAt_view : public view_base {
+public:
+    class iterator;
+    using startAt_func_t = std::function<bool(typename View::value_type&)>;
```

Comment:
```suggestion
    using startAt_func_t = std::function<bool(typename View::value_type const&)>;
```
I've done this also for the other views, since this is a mere predicate only intended to check a condition.